_posts/2023-08-29-what-in-the-ml-is-going-on-around-here.md

```diff
@@ -0,0 +1,473 @@
+
+Moving forward I think I'll stick with a pre-trained model. I'm going to delete all my tasks and models and model groups and start again now that we've learned a few lessons. 
+
+---
+
+## Scene 1 Act 2: Create a Model Group and Register a Model
+
+Let's try again. This time I want to create a model group for all of the models I upload to live under. Check this out: 
+
+```json
+POST /_plugins/_ml/model_groups/_register
+{
+    "name": "i_eat_pieces_of_ml_model_groups_like_you_for_breakfast",
+    "description": "This is Nate trying to make sense of the world."
+}
+
+# Response
+{
+  "model_group_id": "HeO3j4oBKue4OlrZrAL4",
+  "status": "CREATED"
+}
+```
+
+Ok, let's try to register a pre-trained model in that model group by passing the `model_group_id`.
+```json
+POST /_plugins/_ml/models/_register
+{
+    "name": "huggingface/sentence-transformers/all-MiniLM-L12-v2",
+    "model_group_id":"HeO3j4oBKue4OlrZrAL4",
+    "version": "1.0.1",
+    "model_format": "TORCH_SCRIPT"
+}
+
+# Response
+{
+  "task_id": "Q-PAj4oBKue4OlrZ-gLf",
+  "status": "CREATED"
+}
+
+# Task Status
+{
+    "model_id": "R-PAj4oBKue4OlrZ_QJ_",
+    "task_type": "REGISTER_MODEL",
+    "function_name": "TEXT_EMBEDDING",
+    "state": "COMPLETED",
+    "worker_node": [
+    "_QJb--HRS2-7lfq5DCWMiQ"
+    ],
+    "create_time": 1694628903647,
+    "last_update_time": 1694628916260,
+    "is_async": true
+}
+```
+
+I would consider that a great success. We've registered a model with OpenSearch (that is to say, the model exists in the OpenSearch model index), but according to the [ml-framework](https://opensearch.org/docs/latest/ml-commons-plugin/ml-framework/) docs we still need to load the model. This is the process of loading the model into memory.  I've got a model id - let's try it. 
+
+```json
+POST /_plugins/_ml/models/R-PAj4oBKue4OlrZ_QJ_/_load
+
+# Response
+{
+    "task_id": "yeOukIoBKue4OlrZ6gMO",
+    "status": "CREATED"
+}
+
+# Task Details
+{
+    "model_id": "R-PAj4oBKue4OlrZ_QJ_",
+    "task_type": "DEPLOY_MODEL",
+    "function_name": "TEXT_EMBEDDING",
+    "state": "COMPLETED",
+    "worker_node": [
+    "_QJb--HRS2-7lfq5DCWMiQ"
+    ],
+    "create_time": 1694644496910,
+    "last_update_time": 1694644503323,
+    "is_async": true
+}
+```
+
+Success again. Now, the million dollar question. How do we take our own text and documents and have them vectorized when we ingest? 
+
+## The Neural Search Plugin
+
+The [Neural Search Plugin](https://opensearch.org/docs/latest/search-plugins/neural-search/) is the solution. It will translate between vectors and text during ingestion and search time. As our goal is to ingest our own source of text and that's it, it looks like we're getting close to our goal. 
```

Comment:
```suggestion
The [Neural Search plugin](https://opensearch.org/docs/latest/search-plugins/neural-search/) is the solution. It will translate between vectors and text during ingestion and search time. As our goal is to ingest our own source of text and that's it, it looks like we're getting close to our goal. 
```